### PR TITLE
fixbug for finalizer_example

### PR DESCRIPTION
**Pull request by SunhaoKim**

### CODE REVIEW

**Comment by SunhaoKim** [comment]
> 感谢您的拉取请求。在我们查看您的拉取请求之前，您需要签署一份贡献者许可协议 (CLA)。
> 
> 📝 **请按照https://git.k8s.io/community/CLA.md#the-contributor-license-agreement上的说明签署 CLA。**
> 
> CLA 签名可能需要几分钟才能完全注册；之后，请在此处回复一条新评论，我们将进行验证。谢谢。
> 
> * 如果您已经签署了 CLA，则可能我们没有您的 GitHub 用户名，或者您使用的是其他电子邮件地址。检查您现有的 CLA 数据并验证您的[电子邮件是否设置在您的 git commits 上](https://help.github.com/articles/setting-your-email-in-git/)。
> * 如果您作为公司签署了 CLA，请使用您组织的凭据登录https://identity.linuxfoundation.org/projects/cncf以获得授权。
> * 如果您已完成上述操作，但仍遇到报告为未签名的 CLA 的问题，请在 Linux 基金会帮助台上登记一张票：[https](https://support.linuxfoundation.org/) : [//support.linuxfoundation.org/](https://support.linuxfoundation.org/)
> * 如果您在 Linux Foundation Helpdesk 中遇到任何问题，请向备用电子邮件支持地址发送消息：[login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)
> 
> [此处](https://git.k8s.io/community/contributors/guide/pull-requests.md)提供了有关使用 PR 评论与我互动的说明。如果您对我的行为有任何疑问或建议，请针对[kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:)存储库提出问题。我了解[此处](https://go.k8s.io/bot-commands)列出的命令。
t

**Comment by SunhaoKim** [comment]
init cla

File: docs/book/src/cronjob-tutorial/testdata/finalizer_example.go [hunk 1]
```diff
@@ -76,7 +75,7 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// The object is being deleted
 		if controllerutil.ContainsFinalizer(cronJob, myFinalizerName) {
 			// our finalizer is present, so lets handle any external dependency
-			if err := r.deleteExternalResources(cronJob); err != nil {
+			if err := r.delete(ctx, cronJob); err != nil {
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/finalizer_example.go:78]: The idea is to show that by using the finalizer you can do operations before effectively deleting the resource.
So, I agree the example is not clear. What about  

```suggestion
                         // The following method ilustrate that by using the Finalizer 
                         // before we allow the removal of the CronJob 
                         // we can perform all required operations.
                         // e.g do a HTTP request or update any other required 
                         // resource  
			if err := r. deleteExternalResources(); err != nil {
			       return ctrl.Result{}, err
			}
			// After we do all required operations
			// and ensure that the required criteria
			// to delete is succffuly achieved then,
			// we can remove the finalizer and delete it. 
			// Note that you do not need to use the finalizer
			// to remove any resource which is owned by the
			// kind. Remember that we use by controllerutil.<method to set the ownership>
			// More info: <add here the link for the k8s docs that explain the cascate delation>
			....
```
Also, we can improve the following method to better demostrate it::  

```go
func (r *Reconciler) deleteExternalResources() error {
	//
	// delete any external resources associated with the cronJob
	//
	// Ensure that delete implementation is idempotent and safe to invoke
	// multiple times for same object.
}
```

Comment (SunhaoKim) [reply docs/book/src/cronjob-tutorial/testdata/finalizer_example.go]: ok iagree

File: docs/book/src/cronjob-tutorial/testdata/finalizer_example.go [hunk 2]
```diff
@@ -75,12 +75,22 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 			}
-
+			// After we do all required operations
+			// and ensure that the required criteria
+			// to delete is succffuly achieved then,
+			// we can remove the finalizer and delete it.
+			// Note that you do not need to use the finalizer
+			// to remove any resource which is owned by the
+			// kind. Remember that we use by controllerutil.<method to set the ownership>
+			// More info: <add here the link for the k8s docs that explain the cascate delation>
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/finalizer_example.go:93]: See that is missing to shape the msg ^

Comment (SunhaoKim) [reply docs/book/src/cronjob-tutorial/testdata/finalizer_example.go]: ok i commit this

File: docs/book/src/cronjob-tutorial/testdata/finalizer_example.go [hunk 3]
```diff
@@ -89,8 +89,8 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			// we can remove the finalizer and delete it.
 			// Note that you do not need to use the finalizer
 			// to remove any resource which is owned by the
-			// kind. Remember that we use by controllerutil.<method to set the ownership>
-			// More info: <add here the link for the k8s docs that explain the cascate delation>
+			// kind. Remember that we use by controllerutil.RemoveFinalizer
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/finalizer_example.go:92]: controllerutil.RemoveFinalizer -> it is not the method that we set the ownership

Comment (SunhaoKim) [reply docs/book/src/cronjob-tutorial/testdata/finalizer_example.go:92]: ok

Comment (SunhaoKim) [reply docs/book/src/cronjob-tutorial/testdata/finalizer_example.go]: i commit this please  review

File: docs/book/src/cronjob-tutorial/testdata/finalizer_example.go [hunk 4]
```diff
@@ -76,12 +75,22 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 				return ctrl.Result{}, err
 			}
-
+			// After we do all required operations
+			// and ensure that the required criteria
+			// to delete is succffuly achieved then,
+			// we can remove the finalizer and delete it.
+			// Note that you do not need to use the finalizer
+			// to remove any resource which is owned by the
+			// kind. Remember that we use by  it is not the method that we set the ownership
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/finalizer_example.go:92]:
```suggestion
			// kind. Remember that we use by it is not the method that we set the ownership
```

**Review by camilamacedo86 — COMMENTED** [review]
Thank you for your contribution 🥇 
Just a few nits to get it merged

a) Why the import `"k8s.io/kubernetes/pkg/apis/batch"` was removed? 
b) Just a space
c) you just assign the CLA
d) Could you please rebase all commits?
e) Could you please add the mojo `:book:`. See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/CONTRIBUTING.md#pr-process

Thank you .

**Pull request closed without merging**